libs/analytics/src/gtm/initGtm.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * GTM initialization module
+ * Provides type-safe initialization of Google Tag Manager
+ */
+
+import { CowAnalyticsGtm } from './CowAnalyticsGtm'
+
+import { CowAnalytics } from '../CowAnalytics'
+
+declare global {
+  interface Window {
+    dataLayer: unknown[]
+  }
+}
+
+const DEFAULT_GTM_ID = 'GTM-TBX4BV5M'
+
+/**
+ * Initialize GTM and return a CowAnalytics instance
+ * @param gtmId - Optional GTM container ID
+ * @returns CowAnalytics instance backed by GTM
+ */
+export function initGtm(gtmId: string = DEFAULT_GTM_ID): CowAnalytics {
```

Comment:
This function creates a new instance of `CowAnalyticsGtm` on each call and inserts the GTM script in DOM. I see you call the function in many components in cow-fi. We should not do that.
This function should insert the script and instantiate CowAnalyticsGtm only once.

Reply:
Thanks for catching this. I've refactored the `initGtm` function to use a singleton pattern that ensures GTM is only initialized once, regardless of how many times it's called. The script injection now happens only on first initialization, and subsequent calls will return the existing instance. I've also added proper error handling for edge cases like attempting to initialize with different GTM IDs.